dashboard/client/src/pages/serve/ServeReplicaDetailPage.tsx

```diff
@@ -0,0 +1,171 @@
+import Loading from "../../components/Loading";
+import { MetadataSection } from "../../components/MetadataSection";
+import { StatusChip } from "../../components/StatusChip";
+import { ServeDeployment, ServeReplica } from "../../type/serve";
+import { MainNavPageInfo } from "../layout/mainNavContext";
+import { useServeReplicaDetails } from "./hook/useServeApplications";
+
+export const ServeReplicaDetailPage = () => {
+  const { applicationName, deploymentName, replicaId } = useParams();
+
+  const { loading, application, deployment, replica, error } =
+    useServeReplicaDetails(applicationName, deploymentName, replicaId);
+
+  if (error) {
+    return <Typography color="error">{error.toString()}</Typography>;
+  }
+
+  if (loading) {
+    return <Loading loading />;
+  } else if (!replica || !deployment || !application) {
+    return (
+      <Typography color="error">
+        {applicationName} / {deploymentName} / {replicaId} not found.
+      </Typography>
+    );
+  }
+
+  const appName = application.name ? application.name : "-";
+  const {
+    replica_id,
+    state,
+    actor_id,
+    actor_name,
+    node_id,
+    node_ip,
+    pid,
+    start_time_s,
+  } = replica;
+  return (
+    <div>
+      <MainNavPageInfo
+        pageInfo={{
+          id: "serveReplicaDetail",
+          title: replica_id,
+          path: `/serve/applications/${encodeURIComponent(
+            appName,
+          )}/${encodeURIComponent(deployment.name)}/${encodeURIComponent(
+            replica_id,
+          )}`,
+        }}
+      />
+      <MetadataSection
+        metadataList={[
+          {
+            label: "ID",
+            content: {
+              value: replica_id,
+              copyableValue: replica_id,
+            },
+          },
+          {
+            label: "State",
+            content: <StatusChip type="serveReplica" status={state} />,
+          },
+          {
+            label: "Logs",
+            content: (
+              <ServeReplicaLogsLink replica={replica} deployment={deployment} />
+            ),
+          },
+          {
+            label: "Actor ID",
+            content: {
+              value: actor_id ? actor_id : "-",
+              copyableValue: actor_id ? actor_id : undefined,
+              link: actor_id ? generateActorLink(actor_id) : undefined,
+            },
+          },
+          {
+            label: "Actor name",
+            content: {
+              value: actor_name,
+              copyableValue: actor_name,
+            },
+          },
+          {
+            label: "Node ID",
+            content: {
+              value: node_id ? node_id : "-",
+              copyableValue: node_id ? node_id : undefined,
+              link: node_id ? generateNodeLink(node_id) : undefined,
+            },
+          },
+          {
+            label: "Node IP",
+            content: {
+              value: node_ip ? node_ip : "-",
+              copyableValue: node_ip ? node_ip : undefined,
```

Comment:
nit: `node_ip ?? "-"`, also the "-" logic ternary logic could be removed b/c it's handled in

https://github.com/ray-project/ray/blob/74af29a912e156bd5f4c8edcb90733c9ca1625f0/dashboard/client/src/components/MetadataSection/MetadataSection.tsx#L103

we would just have to loosen these types to be optional like ?

https://github.com/ray-project/ray/blob/74af29a912e156bd5f4c8edcb90733c9ca1625f0/dashboard/client/src/components/MetadataSection/MetadataSection.tsx#L17-L19

Reply:
?? doesn't work because it's null not undefined.

I'll update MetadataSection in a follow-up PR to support null or undefined.